components/FutureGraph.tsx

```diff
@@ -0,0 +1,318 @@
+              fill="#FFD053"
+            />
+          </svg>
+        );
+      }
+    }
+
+    return null;
+  };
+
+  /**
+   * Generates the gradient for the graph based on the threshold
+   * @returns SVG gradient
+   */
+  const generateGraphGradient = () => {
+    if (!isLoading && data) {
+      const aboveThreshold = data.forecast_values.some(
+        (forecast) => forecast.expected_generation_kw > graphThreshold
+      );
+
+      if (aboveThreshold) {
+        let gradientPercentage = -107.5 * graphThreshold + 99.25;
```

Comment:
Perhaps separate magic numbers into variables.... but this function is already pretty clean and again there's not much that can make these less magic, they're just styles I guess.